### PR TITLE
guide: clarify deps/outs subfields in DVC files

### PR DESCRIPTION
**Pull request by jorgeorpinel**
~~I just realized we may be missing this section (similar to https://dvc.org/doc/user-guide/project-structure/pipelines-files#output-subfields).~~

This ended up being mainly related to external data (imports).

### CODE REVIEW

File: content/docs/user-guide/project-structure/pipelines-files.md [hunk 1]
```diff
@@ -367,9 +367,21 @@ validation and auto-completion.
...
+> [dependency entries](/doc/user-guide/project-structure/dvc-files#dependency-entries).
+
+| Field                           | Description                                                                                                                                                                                                                                                                                                                           |
+| ------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `md5`<br/>`etag`<br/>`checksum` | Hash value for the file or directory being tracked with DVC. MD5 is used for most locations (local file system and SSH); [ETag](https://en.wikipedia.org/wiki/HTTP_ETag#Strong_and_weak_validation) for HTTP, S3, or Azure external `deps`; and a special _checksum_ for HDFS and WebHDFS. See `dvc import-url` for more information. |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/pipelines-files.md:379]: Hm.. Turns out this is not the case. Not sure why I thought so... Reverting ⌛

File: content/docs/user-guide/project-structure/dvc-files.md [hunk 2]
```diff
@@ -69,18 +69,18 @@ The following subfields may be present under `outs` entries:
...
+| ------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `path`                          | (Required) Path to the dependency (relative to `wdir`, which defaults to the file's location)                                                                                                                                                                                                             |
+| `md5`<br/>`etag`<br/>`checksum` | Only in <abbr>external dependencies</abbr> created with `dvc import-url`: Hash value of the imported file or directory. MD5 is used for most locations; [ETag](https://en.wikipedia.org/wiki/HTTP_ETag#Strong_and_weak_validation) for HTTP, S3, or Azure; and a special _checksum_ for HDFS and WebHDFS. |
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/dvc-files.md:75]: > MD5 is used for most locations

Does that mean specifically local, SSH and GCS? Cc @efiop thanks (just double checking) 🙏

File: content/docs/user-guide/project-structure/pipelines-files.md [hunk 3]
```diff
@@ -369,7 +369,7 @@ validation and auto-completion.
 
 ### Output subfields
 
-> Notice that these are a subset of those in `.dvc` file
+> These include a subset of the fields in `.dvc` file
 > [output entries](/doc/user-guide/project-structure/dvc-files#output-entries).
```
Comment (jorgeorpinel) [inline content/docs/user-guide/project-structure/pipelines-files.md:373]: No longer exclusive bc/ dvc.yaml `deps` now support `checkpoints`.

File: content/docs/user-guide/project-structure/dvc-files.md (hunk 2 above)
Comment (efiop) [reply content/docs/user-guide/project-structure/dvc-files.md:75]: gcs uses etag too.

**Review by efiop — APPROVED** [review]

File: content/docs/user-guide/project-structure/dvc-files.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/user-guide/project-structure/dvc-files.md]: OK, updated.

**Pull request merged**